### PR TITLE
🧪 Sentry: [test coverage improvement] for DatabaseConfig validate

### PR DESCRIPTION
**Pull request by madmax983**
🎯 Target
- Testing gap identified in `DatabaseConfig::validate` inside `autumn/src/config.rs:797`.
- Existing tests checked some valid paths but didn't exhaustively check various edge cases like upper casing, empty strings, and trailing slashes.

💣 Risk
- The validation logic parses database connection strings without thoroughly testing all formatting corner cases that might break standard parsing or lead to invalid credentials being passed down.

🧪 Strategy
- Added `database_validate_url_edge_cases` test inside `mod tests` block.
- The test iterates over a suite of malformed URLs including upper case POSTGRES schemes, blank paths, and non-sensical strings.

🔬 Verification
- Tests successfully pass using `cargo test -p autumn-web --lib config::tests`.

---
*PR created automatically by Jules for task [4958754197653839520](https://jules.google.com/task/4958754197653839520) started by @madmax983*

### CODE REVIEW

File: autumn/src/config.rs.orig [hunk 1]
```diff
@@ -0,0 +1,2420 @@
+//! Framework configuration with sensible defaults and profile-based layering.
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/src/config.rs.orig]: > **P2 Badge Remove the duplicated config source snapshot**
> 
> `autumn/src/config.rs.orig` is a full 2,400+ line copy of `config.rs` that is not part of the module graph, so it becomes a second, stale source of truth immediately after the next real config change. Keeping this file in-tree makes code search/review error-prone (contributors can edit or reference the wrong file) and adds substantial maintenance noise in a commit that otherwise only adds a small test.
> 
> Useful? React with 👍 / 👎.

@jules
